_includes/header.html

```diff
@@ -3,9 +3,10 @@
   <div class="wrapper">
     {%- assign default_paths = site.pages | map: "path" -%}
     {%- assign page_paths = site.header_pages | default: default_paths -%}
+    {%- assign titles_size = site.pages | map: 'title' | join: '' | size -%}
```

Comment:
What if a page without title is added in `header_pages`?

Reply:
It would be joined as an empty string and would not affect the the joined string size `titles_size.` Either way, as the code is right now, a page with no title would not show in the nav either since you have `{%- if my_page.title -%}` further down in the code, which I have also changed to check the joined string size `titles_size`.